libsolidity/codegen/ir/IRGeneratorForStatements.cpp

```diff
@@ -179,6 +180,67 @@ bool IRGeneratorForStatements::visit(FunctionCall const& _functionCall)
 	return false;
 }
 
+// Preliminary implementation that doesn't take care of many things like
+// name collisions etc. Mainly used for test cases.
+bool IRGeneratorForStatements::visit(InlineAssembly const& _inlineAsm)
+{
+	auto const& loc = _inlineAsm.location();
+
+	solAssert(loc.start != 1 && loc.end != -1, "Expected source location");
+	solAssert(!!loc.source, "Expected source location");
+
+	vector<string> params, retParams, args;
+	vector<map<string, string>> assignVars;
+	// Prepare parameters and arguments
+	for (auto const& pair: _inlineAsm.annotation().externalReferences)
+	{
+		auto varDecl = dynamic_cast<VariableDeclaration const*>(pair.second.declaration);
+		solUnimplementedAssert(varDecl, "");
+		solUnimplementedAssert(
+			pair.second.isOffset == false && pair.second.isSlot == false,
+			""
+		);
+
+		args.emplace_back(m_context.variableName(*varDecl));
+		params.emplace_back(varDecl->name());
+		retParams.emplace_back("ret_" + varDecl->name());
+		assignVars.emplace_back(
+			map<string, string>{
+				{"retParam", retParams.back()},
+				{"param", params.back()}
+			}
+		);
+	}
+
+	// Skip the "assembly" literal
+	int blockStart = loc.start + string("assembly").size();
+	solAssert(loc.end > blockStart, "");
+
+	stringstream funcName;
+
+	funcName << "inline$assembly$" << _inlineAsm.id();
+
+	Whiskers inlineAsm(R"(
+		function <funcName> (<params>) -> <retParams> {
```

Comment:
Note that functions do not really generate an independent scope. Identifiers from outside the function are not accessible but they are still visible and will clash with identifiers inside the function (there was a bug that concealed this behaviour in Solidity that was fixed 2 weeks ago).